### PR TITLE
[21-0966] Add logic to automatically check ITF statuses and create ITFs

### PR DESCRIPTION
**Pull request by ndsprinkle**
## Summary
These changes to 21-0966 apply ONLY to the veteran flow.

This PR adds new logic for when a user selects the "veteran" option on the first question. Following that question, a request is made to the intent to file endpoint to check if the veteran has any active ITFs. If the user has two active ITFs (compensation and pension), then the user bypasses the entire form and is taken to the confirmation page. If the user has one active ITF (compensation OR pension), then the user is taken to a benefit selection question where they are asked if they want to apply for the other available benefit. If the user has NO active ITFs, then the usual benefit question is shown with the options to apply for compensation and/or pension.

Following the benefit selection questions, a second request is made to the intent to file endpoint that attempts to create an ITF for the benefits selected. If that creation is successful, then the user is taken to the confirmation page. If that request/creation FAILS, then the user is taken to the next question in the form and will need to continue through the form to the review page in order to submit their ITF.

Here's what's not completely complete as of this PR:

- The confirmation page doesn't always show the correct messaging (need changes on the front end for when both ITFs are already active and when selecting "No" on the individual benefit selection question AND need changes on the back end to support responding to the new individual benefit selection questions)
- The E2E flows do not support the new individual benefit questions or the values set in form data after making the ITF requests
- The tests do not support mocking the intent to file requests
- The loading pages after making a request are not centered and don't display the correct messages (fun forms library additional routes challenges)
- The second loading page looks the same as the first loading page even though it should have different content

## Related issue(s)
department-of-veterans-affairs/va.gov-team-forms#966

## Screenshots
TBD

### CODE REVIEW

File: src/applications/simple-forms/21-0966/config/form.js [hunk 1]
```diff
@@ -98,6 +117,44 @@ const formConfig = {
               : undefined,
           uiSchema: preparerIdentification.uiSchema,
           schema: preparerIdentification.schema,
+          onNavForward: ({ formData, goPath, goNextPath, setFormData }) => {
+            if (preparerIsVeteran({ formData })) {
+              goPath('get-itf-status');
+
+              apiRequest('/intent_to_file')
```
Comment (Thrillberg) [inline src/applications/simple-forms/21-0966/config/form.js:124]: Does this just work? It can figure out the ICN on its own? That's great if so!!

**Review by Thrillberg — APPROVED** [review]
Looks fine to me! Let's try this puppy out in staging!

File: src/applications/simple-forms/21-0966/config/form.js [hunk 2]
```diff
@@ -128,10 +185,94 @@ const formConfig = {
...
+          title: formData => benefitSelectionChapterTitle({ formData }),
+          uiSchema: veteranBenefitSelectionCompensation.uiSchema,
+          schema: veteranBenefitSelectionCompensation.schema,
+          onNavForward: ({ formData, goPath, setFormData }) => {
+            if (formData.benefitSelectionCompensation) {
+              goPath('get-itf-status');
+
+              apiRequest('/intent_to_file/compensation', { method: 'POST' })
+                .then(() => {
+                  goPath('confirmation');
```
Comment (Thrillberg) [inline src/applications/simple-forms/21-0966/config/form.js:237]: Do we need to set the expiration date or anything that gets returned or did we settle on just displaying one year from "now" and ignoring any response from the API?

Comment (ndsprinkle) [reply src/applications/simple-forms/21-0966/config/form.js:237]: That's a good point. I could probably grab the data from the response here and set it to something to display on the confirmation page. I'll do that when I loop back through all of this and fix the UI

**Pull request merged**